src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/metrics/cyclomaticcomplexity/InputCyclomaticComplexityWhenSwitchAsSinglePoint.java

```diff
@@ -0,0 +1,39 @@
+/*
+CyclomaticComplexity
+max = 0
+switchBlockAsSingleDecisionPoint = true
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.metrics.cyclomaticcomplexity;
+
+public class InputCyclomaticComplexityWhenSwitchAsSinglePoint {
+
+   // violation below, 'Cyclomatic Complexity is 5 (max allowed is 0)'
+   void testIf(Object o) {   // 1, function declaration
+      if (o instanceof Integer i && i == 9) {}  // 2, if - 3, &&
+      else if (o instanceof String s && s.length() == 9) {} // 4, if - 5, &&
+   }
+
+   // violation below, 'Cyclomatic Complexity is 4 (max allowed is 0)'
+   void testSwitch(Object o) {   // 1, function declaration
+      switch (o) {  // 2, switch
+         case Integer i when i == 9 -> {} // 3, when
+         case String s when s.length() == 9 -> {} // 4, when
+         default -> {}
+      }
+   }
+
+   // violation below, 'Cyclomatic Complexity is 5 (max allowed is 0)'
+   void testSwitch2(Object o) {  // 1, function declaration
+     switch (o) {    // 2 switch
+         case Integer i -> {}
+         case Point(int x, int y) when (x == 0 && y == 0) -> {} // 3, when - 4, &&
+         case Point(int x, int y) when x == 0 -> {} // 5, when
+         default ->  {}
+      }
+   }
```

Comment:
I am not sure about it, but shouldn't it be more reasonable if we ignore everything inside the switch when `switchBlockAsSingleDecisionPoint` is true? because basically, we should treat switch as a single decision point so considering `when` and `&&` as decision points and adding 1 to them is odd to me when this option is true

Reply:
Please add traditional switches in here with nested `if` statements to allow us to compare how this check handles them

Reply:
done, it got counted. 

https://github.com/checkstyle/checkstyle/blob/87d431a7a0f438e5a858cf11ff3acf025148f936/src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/CyclomaticComplexityCheck.java#L256-L259

looks like we only ignore `case` when this option is true but anything else inside the switch will be counted